vector/src/main/java/im/vector/app/features/analytics/impl/DefaultVectorAnalytics.kt

```diff
@@ -226,9 +229,38 @@ class DefaultVectorAnalytics @Inject constructor(
         return nonNulls
     }
 
+    /**
+     * Adds super properties to the actual property set.
+     * If a property of the same name is already on the reported event it will not be overwritten.
+     */
+    private fun Map<String, Any>.withSuperProperties(): Map<String, Any> {
+        val withSuperProperties = this.toMutableMap()
+        val superProperties = this@DefaultVectorAnalytics.superProperties?.getProperties()
+        superProperties?.forEach {
+            if (!withSuperProperties.containsKey(it.key)) {
+                withSuperProperties[it.key] = it.value
+            }
+        }
+        return withSuperProperties
+    }
+
     override fun trackError(throwable: Throwable) {
         sentryAnalytics
                 .takeIf { userConsent == true }
                 ?.trackError(throwable)
     }
+
+    override fun updateSuperProperties(updatedProperties: SuperProperties) {
+        if (this.superProperties == null) {
+            this.superProperties = updatedProperties
+            return
+        }
+
+        this.superProperties = SuperProperties(
+                platformCodeName = updatedProperties.platformCodeName ?: this.superProperties?.platformCodeName,
+                cryptoSDK = updatedProperties.cryptoSDK ?: this.superProperties?.cryptoSDK,
+                appPlatform = updatedProperties.appPlatform ?: this.superProperties?.appPlatform,
+                cryptoSDKVersion = updatedProperties.cryptoSDKVersion ?: superProperties?.cryptoSDKVersion
+        )
+    }
```

Comment:
I think this can be simplified with:

```kotlin
    override fun updateSuperProperties(updatedProperties: SuperProperties) {
        superProperties = SuperProperties(
                platformCodeName = updatedProperties.platformCodeName ?: superProperties?.platformCodeName,
                cryptoSDK = updatedProperties.cryptoSDK ?: superProperties?.cryptoSDK,
                appPlatform = updatedProperties.appPlatform ?: superProperties?.appPlatform,
                cryptoSDKVersion = updatedProperties.cryptoSDKVersion ?: superProperties?.cryptoSDKVersion
        )
    }
```